### DISCUSSION TITLE
Error when install silo with spack

### DISCUSSION THREAD

**Discussion by lwJi** [LLNL/Silo#365] [Q&A]
I got the following error when I tried to install silo with spack
`H5FDsilo.c:531:5: error: incompatible function pointer types initializing 'haddr_t (*)(const H5FD_t *, H5FD_mem_t)' (aka 'unsigned long (*)(const struct H5FD_t *, enum H5F_mem_t)') with an expression of type 'haddr_t (const H5FD_t *)' (aka 'unsigned long (const struct H5FD_t *)') [-Wincompatible-function-pointer-types]`

the command I used is `spack install silo +hdf5 %oneapi@2023.1.0`

**Comment by lwJi** [comment]
I can go around it by adding `cflags=-Wno-incompatible-function-pointer-types`

**Reply by markcmiller86** [reply]
If feels like you've got a silo + hdf5 version mix problem. What versions of Silo and HDF5 are you getting? I suspect you are still getting old Silo with new HDF5.

What version of spack were you running?

The `H5Dsilo.c` file defines a VFD for HDF5 and is not likely to be actually used by you unless you are running on BG/Q type architectures or unless you explicitly request the `HDF5_SILO` VFD during DBCreate calls.

**Reply by lwJi** [reply]
silo@4.10.2 ^hdf5@1.10.9

for spack, I'm using releases/v0.20

are you saying I should `spack install silo %oneapi@2023.1.0` instead of `spack install silo +hdf5 %oneapi@2023.1.0`

**Reply by markcmiller86** [reply]
Well, do you in fact *know* you need Silo's HDF5 driver? I mean, you may...but you may not...in which case the *default* PDB driver is fin....in which case `~hdf5` might be fine to use.

I don't think Silo 4.10.2 is compatible with hdf5 1.10.9.

Maybe try `^hdf5@1.8.22`

**Reply by markcmiller86** [reply]
Or `+hdf5@1.8.22`

**Reply by lwJi** [reply]
Thanks.
Is silo@4.11 compatible with hdf5@1.10.9?

**Reply by markcmiller86** [reply]
yes
